Added autocomplete results to state

JasonStoltz opened the pull request:
https://github.com/elastic/search-ui/issues/105

- Added an option to `SearchBox` called `autocompleteResults`.
- Added a flag to `setSearchTerm` called `autocompleteResults`.
- Added a new connector method called `autocompleteResults`.
- Added a new state property called `autocompletedResults` (Note the "d").

When the `autocompelteResults` flag is true, a new API request will be invoked to fetch results for and populate `autocompletedResults`.

So basically, if you fire up the sandbox application and start typing, since the `autocompleteResults` flag is set to `true`, you will see API requests being fired off, and the results will actually be populated in state.

This makes them available for us to build a view to consume them, which will be added in a future PR.

![latest](https://user-images.githubusercontent.com/1427475/54057704-ac7ede80-41c1-11e9-8ad5-4130e30a6273.gif)

goodroot commented on packages/search-ui/src/actions/setSearchTerm.js:
```diff
@@ -5,12 +5,13 @@
  *
  * @param searchTerm String
  * @param options Object Additional objects
+ * @param options.autocompleteResults Fetch autocomplete results?
```
Can you please create an issue for autocomplete/debounce documentation? 👍

JasonStoltz replied:
Hey @goodroot. I just added a subtask here: https://github.com/elastic/search-ui/issues/105.

I don't mind updating the README / documentation to include that option, but you're certainly welcome to if you'd prefer.

JasonStoltz commented:
Added an option to SearchBox called autocompleteResults.
Added a flag to setSearchTerm called autocompleteResults.
Added a new connector method called autocompleteResults.
Added a new state property called autocompletedResults (Note the "d").
When the autocompelteResults flag is true, a new API request will be invoked to fetch results for and populate autocompletedResults.

So basically, if you fire up the sandbox application and start typing, since the autocompleteResults flag is set to true, you will see API requests being fired off, and the results will actually be populated in state.

This makes them available for us to build a view to consume them, which will be added in a future PR.
